crypto/ec/ec_lib.c

```diff
@@ -297,12 +359,16 @@ int EC_GROUP_set_generator(EC_GROUP *group, const EC_POINT *generator,
         BN_zero(group->order);
     }
 
-    /* The cofactor is an optional field, so it should be able to be NULL. */
-    if (cofactor != NULL) {
+    /*-
+     * Unfortunately the cofactor is an optional field. So it might be NULL.
+     * Either take the provided positive cofactor, or try to compute it.
+     */
+    if (cofactor != NULL && !BN_is_zero(cofactor) && !BN_is_negative(cofactor)) {
         if (!BN_copy(group->cofactor, cofactor))
             return 0;
-    } else {
-        BN_zero(group->cofactor);
+    }
+    else if (!ec_guess_cofactor(group)) {
```

Comment:
Nit: Put else on same line as }

Reply:
Fixed in c628486f8c59450a6e6722f658dc6ba40743da19 force push